integration_test/tests/power_bi_tests.rs

```diff
@@ -531,4 +531,185 @@ mod integration {
+                FIELD_IDS.iter().for_each(|field_type| {
+                    assert_eq!(
+                        SqlReturn::SUCCESS,
+                        SQLColAttributeW(
+                            stmt as HStmt,
+                            (col_num + 1) as u16,
+                            *field_type,
+                            output_buffer as Pointer,
+                            BUFFER_LENGTH,
+                            str_len_ptr,
+                            numeric_attribute_ptr,
+                        ),
+                        "{}",
+                        get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+                    );
+                    assert_eq!(
+                        expected_col_attr_string_lengths[col_num as usize][field_num], *str_len_ptr,
+                        "mismatch for string_length_ptr value for row:{field_num} col:{col_num}"
+                    );
+                    field_num += 1;
+                });
+            }
+            let expected_get_data_string_lengths = [
+                &[32, -1, 14, 10, 0],
+                &[32, -1, 6, 10, 0],
+                &[32, -1, 6, 8, 0],
+                &[36, -1, 18, 10, 0],
+                &[8, -1, 14, 10, 0],
+                &[8, -1, 10, 10, 0],
+                &[8, -1, 14, 10, 0],
+            ];
+            let mut successful_fetch_count = 0;
+            let str_len_ptr = &mut 0;
+            let mut row_num = 0;
+            loop {
+                let result = SQLFetch(stmt as HStmt);
+                assert!(
+                    result == SqlReturn::SUCCESS || result == SqlReturn::NO_DATA,
+                    "{}",
+                    get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+                );
+                match result {
+                    SqlReturn::SUCCESS => {
+                        successful_fetch_count += 1;
+                        let mut expected_sql_return = SqlReturn::SUCCESS;
+                        for col_num in 0..=4 {
+                            if col_num == 4 {
+                                // REMARKS columns are an empty string, NO_DATA is returned
+                                expected_sql_return = SqlReturn::NO_DATA;
+                            }
+                            assert_eq!(
+                                expected_sql_return,
+                                SQLGetData(
+                                    stmt as HStmt,
+                                    col_num + 1,
+                                    odbc_sys::CDataType::WChar,
+                                    output_buffer as Pointer,
+                                    BUFFER_LENGTH as Len,
+                                    str_len_ptr
+                                ),
+                                "{}",
+                                get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+                            );
+                            assert_eq!(
```

Comment:
Same for this, we have tests for SQLTables which are checking the resultset, so I think you can skip this validation.

Reply:
Sounds good.